### PR TITLE
chore: make `CollectionTtl` pointer and assign default value to  `Ttl` and `RefreshTtl`

### LINKED ISSUE #174: Make CollectionTTL a pointer and try to make behave like our other SDKS when no value passed

### PR DESCRIPTION
**Pull request by poppoerika**
Update the collection operations to accept a `CollectionTtl` pointer as a param.
If no value is passed for `Ttl`, it will default to default ttl that user specified when creating a new cache client.. 
If no value is passed for `RefreshTtl`, it will default to `true`. 

Also updated the new param structure for `NewCacheClient` in the examples.

Closes #174

### CODE REVIEW

**Comment by poppoerika** [comment]
I made both `Ttl` and `RefreshTtl` pointers so that we receive `nil` if users don't pass any values.

File: momento/sorted_set_put.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ type SortedSetPutRequest struct {
 	CacheName string
 	SetName   string
 	Elements  []*SortedSetPutElement
-	Ttl       utils.CollectionTtl
+	Ttl       *utils.CollectionTtl
```
Comment (pgautier404) [inline momento/sorted_set_put.go:32]: I think this is the only place we need to convert anything to a pointer. Then, when it's time to use the CollectionTtl, we can check if it is nil or not. The rest of this should stay as it was.

Comment (poppoerika) [reply momento/sorted_set_put.go:32]: I thought the same thing first but I realized that I cannot determine when `Ttl.refreshTtl` is originally `false` or a user passed `false`.  The same thing for `Ttl.ttl`. Unless a user passes something to `Ttl.ttl` I don't know how we can determine no value was passed to that property. Hence the pointer. If nothing is passed, the "zero" value for a pointer is `nil`. 
I thought if nothing at all is provided for `Ttl`, then `*utils.CollectionTtl` is sufficient but if a user misses one of the properties, we need those to be pointers as well?
Maybe I'm misunderstanding something here. 🤔

Comment (pgautier404) [reply momento/sorted_set_put.go:32]: If the CollectionTtl object is `nil` then the user hasn't passed anything at all, so there's no `ttl` or `refreshTtl` to look at. That's the case we're trying to catch so we can provide the appropriate defaults. The `ttl` and `refreshTtl` properties should stay as values so that users cannot provide nil values for them.

Comment (poppoerika) [reply momento/sorted_set_put.go:32]: Somehow I thought users can pass one of the struct properties 🤷 But got it. Thank you.

File: momento/dictionary_test.go [hunk 2]
```diff
@@ -682,7 +682,7 @@ var _ = Describe("Dictionary methods", func() {
 						DictionaryName: sharedContext.CollectionName,
 						Field:          String("foo"),
 						Value:          String("bar"),
-						Ttl:            utils.CollectionTtl{},
+						Ttl:            &utils.CollectionTtl{},
```
Comment (eaddingtonwhite) [inline momento/dictionary_test.go:685]: maybe just remove this fields from these tests now since we don't want defined for this test case. I think will just read a little more clear and also makes sure we handle not passed value ok. 

Also I think we should add a couple tests to just make sure error gets thrown when we expect when passing a bad value to this.

Comment (poppoerika) [reply momento/dictionary_test.go:685]: Yup, makes sense to me!
I will add some tests.

File: momento/dictionary_increment.go [hunk 3]
```diff
@@ -61,8 +63,9 @@ func (r *DictionaryIncrementRequest) initGrpcRequest(client scsDataClient) error
 		return err
 	}
 
+	collectionTtl := prepareCollectionTtl(r, client.defaultTtl)
 	var ttl uint64
-	if ttl, err = prepareTtl(r, client.defaultTtl); err != nil {
+	if ttl, err = prepareCollectionTtlTtl(collectionTtl.Ttl, client.defaultTtl); err != nil {
```
Comment (schwern) [inline momento/dictionary_increment.go:68]: Previously we had one function that returned the TTL  and refreshTTL values to pass to gRPC. I think that would be simpler here.

File: momento/dictionary_test.go [hunk 4]
```diff
@@ -148,6 +146,18 @@ var _ = Describe("Dictionary methods", func() {
 		Entry("both nil", nil, nil),
 	)
 
+	It("try using negative ttl for set", func() {
```
Comment (schwern) [inline momento/dictionary_test.go:149]: It should read like a sentence describing the expected behavior.

```suggestion
	It("errors with a negative ttl", func() {
```

File: momento/simple_cache_client.go [hunk 5]
```diff
@@ -239,11 +240,15 @@ func (c defaultScsClient) SetAddElements(ctx context.Context, r *SetAddElementsR
...
+	ttl := r.Ttl
+	if r.Ttl == nil {
+		ttl = assignDefaultCollectionTtl(c.dataClient.defaultTtl)
+	}
 	newRequest := &SetAddElementsRequest{
 		CacheName: r.CacheName,
 		SetName:   r.SetName,
 		Elements:  []Value{r.Element},
-		Ttl:       r.Ttl,
+		Ttl:       ttl,
```
Comment (schwern) [inline momento/simple_cache_client.go:251]: Is it necessary to handle a nil TTL? That should be handled by making the SetAddElementsRequest.

I think this can be simplified by calling the client method instead of making a request.

```
_, err := c.SetAddElements(ctx, &SetAddElementsRequest{...})
if err != nil {
  return err
}
return &SetAddElementSuccess
```

File: momento/simple_cache_client.go [hunk 6]
```diff
@@ -348,13 +353,17 @@ func (c defaultScsClient) DictionarySetField(ctx context.Context, r *DictionaryS
 			),
 		)
 	}
+	ttl := r.Ttl
+	if r.Ttl == nil {
+		ttl = assignDefaultCollectionTtl(c.dataClient.defaultTtl)
+	}
```
Comment (schwern) [inline momento/simple_cache_client.go:359]: This should be handled by making the DictionarySetFieldsRequest?

File: momento/dictionary_test.go [hunk 7]
```diff
@@ -672,32 +682,6 @@ var _ = Describe("Dictionary methods", func() {
 			).Error().To(BeNil())
 		})
 
-		When("collection TTL is empty", func() {
```
Comment (schwern) [inline momento/dictionary_test.go:675]: We still need this test, but the CollectionTTL is null.

File: momento/dictionary_increment.go (hunk 3 above)
Comment (schwern) [reply momento/dictionary_increment.go:68]: PS I understand if you don't want to redo all that work.

File: momento/dictionary_test.go (hunk 7 above)
Comment (poppoerika) [reply momento/dictionary_test.go:675]: Ah sorry about that! Thank you for catching this!

**Review by eaddingtonwhite — APPROVED** [review]

**Review by schwern — APPROVED** [review]
LGTM
